extensions/data-transfer/portability-data-transfer-imgur/src/main/java/org/datatransferproject/datatransfer/imgur/photos/ImgurPhotosExporter.java

```diff
@@ -307,9 +333,20 @@ private List<Map<String, Object>> requestData(TokensAndUrlAuthData authData, Str
   }
 
   private InputStream getImageAsStream(String imageUrl) throws IOException {
-    URL url = new URL(imageUrl);
+    // if (url == null) {
+    //   url = from(imageUrl);
+    // }
+    URL url = urlFactory.fromUrl(imageUrl);
     HttpURLConnection conn = (HttpURLConnection) url.openConnection();
     conn.connect();
     return conn.getInputStream();
   }
+
+  // private static URL from(String imageUrl) throws IOException{
+  //   return new URL(imageUrl);
+  // }
+
+  public interface UrlFactory{
```

Comment:
I don't _think_ you need this, btw. I think you can just declare the type of your `urlFactory` param to be `Function<String, URL> urlFactory` in your constructor.

Reply:
changed from interface to declaring the param type, thanks